packages/react-jss/src/createUseStyles.js

```diff
@@ -0,0 +1,148 @@
+// @flow
+
+import React from 'react'
+import isInBrowser from 'is-in-browser'
+import {ThemeContext as DefaultThemeContext} from 'theming'
+
+import JssContext from './JssContext'
+import {
+  createStaticSheet,
+  addDynamicRules,
+  updateDynamicRules,
+  removeDynamicRules
+} from './utils/sheets'
+import {getIndex} from './utils/indexCounter'
+import type {HookOptions, Styles} from './types'
+import {manageSheet, unmanageSheet} from './utils/managers'
+import {getSheetClasses} from './utils/getSheetClasses'
+
+const noTheme = {}
+
+const createUseStyles = <Theme: {}>(styles: Styles<Theme>, options?: HookOptions<Theme> = {}) => {
+  const {index = getIndex(), theming, name = 'Hook', ...sheetOptions} = options
+  const isFirstMount = React.useRef(true)
+  const ThemeContext = (theming && theming.context) || DefaultThemeContext
+  const useTheme =
+    typeof styles === 'function'
+      ? // $FlowFixMe
+        (): Theme => React.useContext(ThemeContext)
+      : // $FlowFixMe
+        (): Theme => noTheme
+
+  const useLayoutEffect = isInBrowser ? React.useLayoutEffect : React.useEffect
+
+  return (data: any) => {
+    const context = React.useContext(JssContext)
+    const theme = useTheme()
+
+    if (context.disableStylesGeneration) {
```

Comment:
Is there some guard in the provider that prevents this value from changing? Otherwise we get an error from react about changing number of hooks calls.

Reply:
We implemented it because someone wanted it to use it in a way where it's not changed. Though there is no rule against changing this. I will move it further down